src/parsers/texttracks/webvtt/__tests__/utils.test.ts

```diff
@@ -0,0 +1,402 @@
+      expect(isStartOfRegionBlock(["REGION"], 0)).to.equal(true);
+    });
+
+    it("should return true if called on line containing `REGION` and spaces", () => {
+      expect(isStartOfRegionBlock(["REGION "], 0)).to.equal(true);
+      expect(isStartOfRegionBlock(["REGION  "], 0)).to.equal(true);
+      expect(isStartOfRegionBlock(["REGION         "], 0)).to.equal(true);
+    });
+
+    /* tslint:disable max-line-length */
+    it("should return true if called on line containing `REGION` and spaces and text", () => {
+    /* tslint:enable max-line-length */
+      expect(isStartOfRegionBlock(["REGION dsj f"], 0)).to.equal(true);
+      expect(isStartOfRegionBlock(["REGION   oej ewj "], 0)).to.equal(true);
+      expect(isStartOfRegionBlock(["REGION         eowj pogj qpeoj"], 0)).to.equal(true);
+    });
+
+    /* tslint:disable max-line-length */
+    it("should return false if called on a line containing `REGION` and text attached", () => {
+    /* tslint:enable max-line-length */
+      expect(isStartOfRegionBlock(["REGIONdsj f"], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["REGIONoej ewj "], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["REGIONeowj pogj qpeoj"], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["REGIONREGION"], 0)).to.equal(false);
+    });
+
+    it("should return false if called on a note block", () => {
+      expect(isStartOfRegionBlock(["NOTE SOMETHING"], 0)).to.equal(false);
+    });
+
+    it("should return false if called on a style block", () => {
+      expect(isStartOfRegionBlock(["STYLE SOMETHING"], 0)).to.equal(false);
+    });
+
+    it("should return false if called on an empty line", () => {
+      expect(isStartOfRegionBlock([""], 0)).to.equal(false);
+    });
+
+    it("should return false for any other cases", () => {
+      expect(isStartOfRegionBlock(["1"], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["ababa abs"], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["a"], 0)).to.equal(false);
+      expect(isStartOfRegionBlock([" "], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["NOTESOMETHING"], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["REGIONSOMETHING"], 0)).to.equal(false);
+      expect(isStartOfRegionBlock(["STYLESOMETHING"], 0)).to.equal(false);
+    });
+  });
+
+  describe("isStartOfStyleBlock", () => {
+    it("should return true if called on a `STYLE` line", () => {
+      expect(isStartOfStyleBlock(["STYLE"], 0)).to.equal(true);
+    });
+
+    it("should return true if called on line containing `STYLE` and spaces", () => {
+      expect(isStartOfStyleBlock(["STYLE "], 0)).to.equal(true);
+      expect(isStartOfStyleBlock(["STYLE  "], 0)).to.equal(true);
+      expect(isStartOfStyleBlock(["STYLE         "], 0)).to.equal(true);
+    });
+
+    /* tslint:disable max-line-length */
+    it("should return true if called on line containing `STYLE` and spaces and text", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 308. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=233&issues=AWOsLbNnmXhJxGlVaMbu&open=AWOsLbNnmXhJxGlVaMbu)

[ISSUE_KEY]: AWOsLbNnmXhJxGlVaMbu